nebula-logger/core/main/logger-engine/classes/Logger.cls

```diff
@@ -2933,19 +2928,6 @@ global with sharing class Logger {
         }
     }
 
-    @TestVisible
-    private static void setMockScenarioRule(LogScenarioRule__mdt scenarioRule) {
-        // TODO Add validation for DeveloperName and Scenario__c fields != null
-        // if (String.isBlank(scenarioRule.DeveloperName) == true) {
-        //     throw new IllegalArgumentException('DeveloperName is required on `LogScenarioRule__mdt: \n' + JSON.serializePretty(scenarioRule));
-        // }
-        // if (String.isBlank(scenarioRule.Scenario__c) == true) {
-        //     throw new IllegalArgumentException('Scenario__c is required on `LogScenarioRule__mdt: \n' + JSON.serializePretty(scenarioRule));
-        // }
-
-        MOCK_SCENARIO_TO_SCENARIO_RULE.put(scenarioRule.Scenario__c, scenarioRule);
-    }
-
     private static SaveMethod getSaveMethod(String saveMethodName) {
         try {
             return SaveMethod.valueOf(saveMethodName);
```

Comment:
I did some profiling on `Enum.valueOf(value)` and found performance pretty wanting for repeated calls to it. Might be a TODO for later but if `getSaveMethod` is being called more than once per transaction it unfortunately might be more efficient to cache a map of string to the individual save methods yourself

Reply:
Ooooh, that's a great idea - I'll add a TODO to get that changed in another branch I'm working on for some optimizations.

Reply:
Nevermind, I'm just going to implement this now 😄 I wrote the TODO in the class locally, but the actual code changes are about as long as the TODO, so I'm just going include the optimization with this PR. The first time that the method is called, it will now cache the enum names & values in a new private map constant `SAVE_METHOD_NAME_TO_SAVE_METHOD`:

```java
private static SaveMethod getSaveMethod(String saveMethodName) {
    if (SAVE_METHOD_NAME_TO_SAVE_METHOD.isEmpty() == true) {
        for (SaveMethod method : SaveMethod.values()) {
            SAVE_METHOD_NAME_TO_SAVE_METHOD.put(method.name(), method);
        }
    }

    return SAVE_METHOD_NAME_TO_SAVE_METHOD(saveMethodName);
}
```